lib/dialogs.dart

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2015, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library dartpad.dialogs;
+
+import 'dart:html';
+
+import 'elements/elements.dart';
+import 'sharing/gists.dart';
+import 'sharing/mutable_gist.dart';
+
+class SharingDialog extends DDialog {
+  final GistContainer gistContainer;
+  final GistController gistController;
+
+  ParagraphElement _text;
+  DButton _cancelButton;
+  DButton _shareButton;
+  DButton _closeButton;
+  DElement _div;
+  DInput _padUrl;
+  DInput _gistUrl;
+
+  SharingDialog(this.gistContainer, this.gistController) : super(title: 'Sharing') {
+    element.classes.toggle('sharing-dialog', true);
+
+    _text = content.add(new ParagraphElement());
+
+    // About to share.
+    _cancelButton = new DButton.button(text: 'Cancel');
+    _cancelButton.onClick.listen((_) => hide());
+    _shareButton = new DButton.button(text: 'Share it!', classes: 'default');
+    _shareButton.onClick.listen((_) => _performShare());
+
+    // Already sharing.
+    _closeButton = new DButton.button(text: 'Close', classes: 'default');
```

Comment:
I like it. I may hold off on doing that for this PR however, I already have a lot of changes in it.